h1b/src/hil/flash/driver.rs

```diff
@@ -24,7 +24,7 @@ pub const WRITE_OPCODE: u32 = 0x27182818;
 /// The H1B flash driver. The hardware interface (either the real flash modules
 /// or the fake) is injected to support testing. This will not configure the
 /// globalsec flash regions -- that must be done independently.
-pub struct Flash<'d, A: Alarm + 'd, H: Hardware<'d> + 'd> {
+pub struct Flash<'d, A: Alarm + 'd, H: Hardware + 'd> {
```

Comment:
Why is it parameterized by Alarm and Hardware? The standard approach in Tock is to only have the lifetime; if fields are references, then the structure doesn't require the actual types.

Reply:
There are a few reasons I preferred this approach:

1. Using dynamic polymorphism instead of static polymorphism implies you need the dynamicism, and we don't.
2. It is probably cheaper (reduces the size of the driver by 1 word and allows inlining of method calls).

That said, I'm not strongly opinionated on the matter and if you still think I should switch to trait objects then I'll do so.